logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueUtils.java

```diff
@@ -41,10 +41,7 @@ static int extractSegmentId(Path p) {
     }
 
     static Stream<Path> listFiles(Path path, String suffix) throws IOException {
-        try(final Stream<Path> files = Files.list(path)) {
-            return files.filter(p -> p.toString().endsWith(suffix))
-                    .collect(Collectors.toList()).stream();
-        }
+        return Files.list(path).filter(p -> p.toString().endsWith(suffix));
```

Comment:
It's not clear to me why the removal of try-finally on a closable resource (the stream of file names) is functional to this fix.
At the end it creates a new stream of paths from  a stream of paths, however the originating stream resource should be always be cleanly closed.

Reply:
You are right. It is a file IO and should be try-finally, will revert it